django/db/models/lookups.py

```diff
@@ -548,3 +548,48 @@ def get_bound_params(self, start, finish):
 class YearLte(YearLookup, LessThanOrEqual):
     def get_bound_params(self, start, finish):
         return (finish,)
+
+
+class UUIDMixin:
+    """Strip hyphens from a value when filtering a UUIDField"""
+
+    def process_rhs(self, qn, connection):
+        rhs, params = super().process_rhs(qn, connection)
+        if not connection.features.has_native_uuid_field:
+            params = [param.replace('-', '') for param in params]
```

Comment:
Should `params` be a tuple? i.e.
```python
            params = tuple(param.replace('-', '') for param in params)
```

Just want to make sure we avoid issues such as the one in #11784.